python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py

```diff
@@ -89,41 +95,87 @@ def _timestamp_type_repair(df, dt_col):
+                            "initialize TSDataset.")
+            return False, intervals
+    else:
+        df_column = df[dt_col]
+        intervals = df_column.shift(-1) - df_column
+        unique_intervals = intervals[:-1].unique()
+        if len(unique_intervals) > 1:
+            logging.warning("There are irregular interval(more than one interval length)"
+                            " among the data. You can call .resample(interval).impute() "
+                            "first to clean the data manually, or set repair=True when "
+                            "initialize TSDataset.")
+            return False, intervals
+        return True, intervals
+
+
+def _time_interval_repair(df, dt_col, intervals, id_col=None):
     '''
     This check is used to get consitent time interval by resample data according to
     the mode of original intervals.
     '''
-    mode = interval[:-1].mode()[0]  # Timedelta
-    from bigdl.chronos.data.utils.resample import resample_timeseries_dataframe
-    # TODO: how to change into inplace modification
-    try:
-        df = resample_timeseries_dataframe(df, dt_col=dt_col,
-                                           interval=mode,
-                                           id_col=id_col)
-        logging.warning(f"Dataframe has be resampled according to interval {mode}.")
-        return df, True
-    except:
-        return df, False
+    if id_col is not None and isinstance(intervals, dict):
+        # multi_id case
+        new_df = []
+        for id_, interval in intervals.items():
+            mode = interval[:-1].mode()[0]  # Timedelta
+            from bigdl.chronos.data.utils.resample import resample_timeseries_dataframe
+            # TODO: how to change into inplace modification
+            df_id = df[df[id_col] == id_]
+            try:
+                df_ = resample_timeseries_dataframe(df_id, dt_col=dt_col,
+                                                    interval=mode,
+                                                    id_col=id_col)
+                logging.warning(f"Dataframe has be resampled according to interval {mode}.")
+                # df[df[id_col] == id_] = df_
+                new_df.append(df_)
+            except:
+                return df, False
+        new_df = pd.concat(new_df, axis=0, ignore_index=True)
+        return new_df, True
+    else:
+        mode = intervals[:-1].mode()[0]  # Timedelta
+        from bigdl.chronos.data.utils.resample import resample_timeseries_dataframe
+        # TODO: how to change into inplace modification
+        try:
+            df = resample_timeseries_dataframe(df, dt_col=dt_col,
+                                               interval=mode,
+                                               id_col=id_col)
+            logging.warning(f"Dataframe has be resampled according to interval {mode}.")
+            return df, True
+        except:
+            return df, False
```

Comment:
please use `df.groupby(...).apply(resample_timeseries_dataframe, ...)` here.

Reply:
Have change to df.groupby.
Acceleration looks like:
![image](https://user-images.githubusercontent.com/105281011/196610507-e2fc26a2-1c02-44b8-a33e-a632e3fa5f71.png)